### PR TITLE
NIAD-809 Adding logging level and add common identifiers to logging format

### PR DESCRIPTION
**Pull request by z-uddin**

### CODE REVIEW

**Review by ElliotB-96 — DISMISSED** [review]
Looks good to me, can also see both conversation id and task id in the logs.

File: service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/ConversationIdService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package uk.nhs.adaptors.gp2gp.common.service;
+
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ConversationIdService {
```
Comment (mattd-kainos) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/ConversationIdService.java:7]: I'd consider making a single MDC service with `applyConversationId` and `applyTaskId` methods and a single `resetAll` method. That encapsulates all of the code related to logging diagnostic identifiers in a single place.

I'd question whether we need getters at all. They were used in NHAIS but I'm not sure that was a great design choice.

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
Looks good, all of the ids are set and reset from the correct places. Recommend having a single service to manage all mapped diagnostic values.

**Review by ElliotB-96 — APPROVED** [review]

**Review by mattd-kainos — APPROVED** [review]
Looks good! Also did a quick test and they values are included in the logs as expected 

```
"mdc":{"ConversationId":"1f712f0c-8d56-40ab-96bc-5eb6a31ab532","TaskId":"696b4617-59c4-44f6-aded-1c40ec012b5b"}
```

**Pull request merged**